TargetLibraries/PULPOpen/inc/mchan_v6.h

```diff
@@ -68,17 +69,20 @@ static void mchan_transfer_2d_ext_strided(uint32_t cmd, void *loc, void *ext,
 static uint32_t mchan_channel_alloc() { return *cmd_ptr; }
 
 static void mchan_channel_free(uint32_t channel_id) {
-  // TODO: assert channel_id is smaller then 32
+  if (channel_id > MCHAN_TRANSFER_ID_MAX)
+    return;
   *status_ptr = 1 << channel_id;
 }
 
 static uint32_t mchan_channel_is_busy(uint32_t channel_id) {
-  // TODO: assert channel_id is smaller then 32
+  if (channel_id > MCHAN_TRANSFER_ID_MAX)
+    return 0;
   return *status_ptr & (1 << channel_id);
 }
 
 static void mchan_channel_wait(uint32_t channel_id) {
-  // TODO: assert channel_id is smaller then 32
+  if (channel_id > MCHAN_TRANSFER_ID_MAX)
+    return;
```

Comment:
I am against silently failing and using that as a feature in the implementation. The design goal of this was to be as bare bones as possible so I wouldn't do this check here but in the code generation where needed.

If we want the check I would prefer it crashes execution or returns an exit status so that the user knows something is wrong.